packages/quantic/force-app/main/default/lwc/quanticSearchBox/quanticSearchBox.html

```diff
@@ -8,21 +8,60 @@
           aria-haspopup="listbox"
           role="combobox"
         >
-          <div class="slds-combobox__form-element" role="none">
-            <input
-              id="combobox-id"
-              class="slds-input slds-combobox__input"
-              placeholder={placeholder}
-              type="search"
-              onkeyup={onKeyup}
-              onfocus={onFocus}
-              onblur={onBlur}
-              aria-autocomplete="list"
-              aria-controls="listbox-id"
-              autocomplete="off"
-              role="textbox"
-            />
-          </div>
+          <template if:true={withoutSubmitButton}>
```

Comment:
Seems to be a lot of repeated code between the two different templates, 

Would you prefer to make a `get searchBoxContainerClass()` JS method that would return the correct class string and then adapt your CSS for that?

And only surround the lightning-icon with the template if:true ?

Reply:
You're right that would probably better. I think I would need another template if:false around the left lightning-icon, and to add a class to the input for the add-on after checking `withoutSubmitButton`. I'll give it a go, thanks!